nntrainer/layers/weight_layer.h

```diff
@@ -95,8 +112,12 @@ class WeightLayer : public LayerImpl {
   static constexpr const char *type = "weight";
 
 private:
-  std::tuple<props::TensorDimension> weight_props;
-  std::array<unsigned int, 1> weight_idx; /**< indices of the weights */
+  std::tuple<std::vector<props::TensorDimension>,
+             std::vector<props::TensorDataType>, std::vector<props::WeightName>>
+    weight_props;
+
+  std::vector<unsigned int> weight_idx; /**< indices of the weights */
+  unsigned int n_weight;
```

Comment:
Wouldn't invariant assert(n_weights == weight_idx.size()) be always true?
Could `n_weights` field be removed in favour of `weight_idx.size()`?